File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;
// 섞은 음식의 스코빌 지수 = 
// 가장 맵지 않은 음식의 스코빌 지수 + (두 번째로 맵지 않은 음식의 스코빌 지수 * 2)

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int, vector<int>, greater<int>> pq;
    
    for(int i = 0; i < scoville.size(); i++){
        pq.push(scoville[i]);
    }
    
    while (pq.size() >= 2 && K > pq.top()){
        int firstMin = pq.top();
        pq.pop();
        int secondMin = pq.top();
        pq.pop();
        pq.push(firstMin + (secondMin*2));
        answer++;
    }
    
    if (!pq.empty() && pq.top() < K) return -1;
    return answer;
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)  
    
    while len(scoville) >= 2 and scoville[0] < K:
        first_min = heapq.heappop(scoville)
        second_min = heapq.heappop(scoville)
        new_scoville = first_min + (second_min * 2)
        heapq.heappush(scoville, new_scoville)
        answer += 1
    
    if scoville[0] < K:
        return -1
    return answer